libpod/runtime_img.go

```diff
@@ -123,8 +124,59 @@ func (r *Runtime) RemoveImage(image *image.Image, force bool) (string, error) {
 		// reponames and no force is applied, we error out.
 		return "", fmt.Errorf("unable to delete %s (must force) - image is referred to in multiple tags", image.ID())
 	}
+	err = image.Remove(force)
+	if err != nil && errors.Cause(err) == storage.ErrImageUsedByContainer {
+		if errStorage := r.rmStorageContainers(force, image); errStorage == nil {
+			// Containers associated with the image should be deleted now,
+			// let's try removing the image again.
+			err = image.Remove(force)
+		} else {
+			err = errStorage
+		}
+	}
+	return image.ID(), err
+}
+
+// Remove containers that are in storage rather than Podman.
+func (r *Runtime) rmStorageContainers(force bool, image *image.Image) error {
 
-	return image.ID(), image.Remove(force)
+	ctrIDs, err := storageContainers(image.ID(), r.store)
+	if err != nil {
+		return errors.Wrapf(err, "error getting containers for image %q", image.ID())
+	}
+
+	if len(ctrIDs) > 0 && force {
+		if err = removeStorageContainers(ctrIDs, r.store); err != nil {
+			return errors.Wrapf(err, "error removing containers %v for image %q", ctrIDs, image.ID())
+		}
+	}
+	return nil
+}
+
+// Returns a list of storage containers associated with the given ImageReference
+func storageContainers(imageID string, store storage.Store) ([]string, error) {
+
```

Comment:
Extra whitespace

Reply:
Fixed, found another rogue one and zapped that too.